src/utils/event-dispatcher.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-unneeded-ternary */
+const helpers = require('../utils/helpers');
+
+// Create `CustomEvent` with fallback
+const createCustomEvent = (eventName, detail) => {
+  try {
+    return new window.CustomEvent(eventName, { detail });
+  } catch (e) {
+    const evt = document.createEvent('CustomEvent');
+
+    evt.initCustomEvent(eventName, false, false, detail);
+
+    return evt;
+  }
+};
+
+class EventDispatcher {
+  constructor(options) {
+    this.events = [];
+    this.enabled = (options && options.enabled === false)
```

Comment:
Would `this.enabled = !!options.enabled` have the same effect?

Reply:
following the other comment i made, this will always be enabled unless otherwise specified. since the EventDispatcher will probably not always be enabled, we might want to go with what zubin suggested and do `!!options.enabled`